Tools/LyTestTools/ly_test_tools/o3de/material_canvas_test.py

```diff
@@ -0,0 +1,174 @@
+        self.extra_cmdline_args = []
+        # Whether to use null renderer, this will override use_null_renderer for the Suite if not None
+        self.use_null_renderer = None
+
+    @staticmethod
+    def setup(instance: MaterialCanvasTestSuite.MultiTestCollector,
+              request: _pytest.fixtures.FixtureRequest,
+              workspace: AbstractWorkspaceManager) -> None:
+        """
+        User-overrideable setup function, which will run before the test.
+        :param instance: Parent MaterialCanvasTestSuite.MultiTestCollector instance executing the test
+        :param request: PyTest request object
+        :param workspace: LyTestTools workspace manager
+        """
+        pass
+
+    @staticmethod
+    def wrap_run(instance: MaterialCanvasTestSuite.MultiTestCollector,
+                 request: _pytest.fixtures.FixtureRequest,
+                 workspace: AbstractWorkspaceManager,
+                 material_canvas_test_results: MultiTestSuite.TestData) -> None:
+        """
+        User-overrideable wrapper function, which will run both before and after test.
+        Any code before the 'yield' statement will run before the test. With code after yield run after the test.
+        Setup will run before wrap_run starts. Teardown will run after it completes.
+        :param instance: Parent MaterialCanvasTestSuite.MultiTestCollector instance executing the test
+        :param request: PyTest request object
+        :param workspace: LyTestTools workspace manager
+        :param material_canvas_test_results: Currently recorded MaterialCanvas test results
+        """
+        yield
+
+    @staticmethod
+    def teardown(instance: MaterialCanvasTestSuite.MultiTestCollector,
+                 request: _pytest.fixtures.FixtureRequest,
+                 workspace: AbstractWorkspaceManager,
+                 material_canvas_test_results: MultiTestSuite.TestData) -> None:
+        """
+        User-overrideable teardown function, which will run after the test
+        :param instance: Parent MaterialCanvasTestSuite.MultiTestCollector instance executing the test
+        :param request: PyTest request object
+        :param workspace: LyTestTools workspace manager
+        :param material_canvas_test_results: Currently recorded MaterialCanvas test results
+        """
+        pass
+
+
+class MaterialCanvasSharedTest(SharedTest):
+    """
+    Test that will run in parallel with tests in different MaterialCanvas instances, as well as serially batching
+    with other tests in each MaterialCanvas instance. Minimizes total test run duration.
+
+    Does not support per test setup/teardown to avoid creating race conditions
+    """
+    # Specifies if the test can be batched in the same MaterialCanvas
+    is_batchable = True
+    # Specifies if the test can be run in multiple MaterialCanvases in parallel
+    is_parallelizable = True
+
+
+class MaterialCanvasParallelTest(MaterialCanvasSharedTest):
```

Comment:
why does this implement MaterialCanvasSharedTest? should this implement ParallelTest?